bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +829,115 @@ describe('ClientEncryption', function () {
+        .catch(error => error);
+      expect(error).to.be.instanceOf(MongoCryptError);
+    });
+
+    it('throws MongoCryptCreateEncryptedCollectionError if createDataKey rejects', async () => {
+      const customError = new Error('evil!');
+      const stub = sinon.stub(clientEncryption, 'createDataKey');
+      const keyId = new Binary(Buffer.alloc(16, 0), 4);
+      stub.onCall(0).resolves(keyId);
+      stub.onCall(1).rejects(customError);
+
+      const error = await clientEncryption
+        .createEncryptedCollection(db, collectionName, {
+          provider: 'local',
+          createCollectionOptions: {
+            encryptedFields: { fields: [{}, {}, { keyId: 'cool id!' }] }
+          }
+        })
+        .catch(error => error);
+
+      expect(error).to.be.instanceOf(MongoCryptCreateEncryptedCollectionError);
+      expect(error.cause).to.equal(customError);
+      expect(error.encryptedFields).property('fields').that.is.an('array');
+      expect(error.encryptedFields.fields).to.have.lengthOf(3);
+      expect(error.encryptedFields.fields).to.have.nested.property('[0].keyId', keyId);
+      expect(error.encryptedFields.fields).to.not.have.nested.property('[1].keyId');
+      expect(error.encryptedFields.fields).to.have.nested.property('[2].keyId', 'cool id!');
+    });
+
+    it('creates keyIds where they are nullish without editing input', async () => {
+      const encryptedFields = {
+        set escCollection(v) {
+          throw new Error('should not modify escCollection');
+        },
+        set eccCollection(v) {
+          throw new Error('should not modify eccCollection');
+        },
+        set ecocCollection(v) {
+          throw new Error('should not modify ecocCollection');
+        },
+        fields: [
+          { keyId: false },
+          {
+            keyId: null,
+            set path(v) {
+              throw new Error('should not modify path');
+            },
+            set bsonType(v) {
+              throw new Error('should not modify bsonType');
+            },
+            set queries(v) {
+              throw new Error('should not modify queries');
+            }
+          }
+        ]
+      };
+
+      const keyId = new Binary(Buffer.alloc(16, 0), 4);
+      sinon.stub(clientEncryption, 'createDataKey').resolves(keyId);
+
+      const result = await clientEncryption.createEncryptedCollection(db, collectionName, {
+        provider: 'local',
+        createCollectionOptions: {
+          encryptedFields
+        }
+      });
+
+      expect(result).to.have.property('namespace', 'createEncryptedCollectionDb.secure');
+      expect(encryptedFields, 'original encryptedFields should be unmodified').nested.property(
+        'fields[0].keyId',
+        false
+      );
+      expect(
+        result.options.encryptedFields,
+        'encryptedFields created by helper should have replaced nullish keyId'
+      ).nested.property('fields[1].keyId', keyId);
```

Comment:
my two cents - mocking a return value and then asserting on properties of your own mocked method is circular.

i think instead we could spy on `db.createCollection` (and return enough just to satisfy the code path, like a Promise.resolve() or something) and then assert that `createCollection` was called with the expected parameters.

Or we could consider running these tests against a live server and not mocking the `Db` object at all.